documentation/SupportingDocuments/CAMARA-Release-Workflow-and-Metadata-Proposal.md

```diff
@@ -0,0 +1,272 @@
+
+release_number: r4.1
+
+release_status: alpha  # must be one of: draft, alpha, rc, release
+
+apis:
+  - name: location-verification
+    target_version: 3.2.0
+    api_status: rc
+    main_contacts:
+      - githubUser1
+      - githubUser2
+
+  - name: location-retrieval
+    target_version: 0.5.0
+    api_status: rc
+    main_contacts:
+      - githubUser3
+
+  - name: some-new-location-service
+    target_version: 0.1.0
+    api_status: alpha
+    main_contacts:
+      - githubUser4
+
+commonalities_version: 1.2.0-rc.1
+identity_consent_management_version: 1.1.0
+```
+
+👉 Notes:
+- All APIs must meet the minimum `api_status` for the current `release_status`.
+- `draft` status can't be released 
+- APIs below the threshold block tagging (enforced via CI).
+- Changes to the file can only be merged into main if the APIs and the repository are fulfilling the intendent status.
+
+### 2. `release-metadata.yaml` (on release branch)
+
+Generated automatically from the release plan and committed before tagging.
+
+```yaml
+release_number: r4.1
+meta_release: Fall26
+release_date: 2025-11-22
+status: alpha
+
+apis:
+  - name: location-verification
+    version: 3.2.0-alpha.1
+
+  - name: location-retrieval
+    version: 0.5.0-alpha.1
+
+  - name: some-new-location-service
+    version: 0.1.0-alpha.1
+
+commonalities_version: 1.2.0-rc.1
+identity_consent_management_version: 1.1.0
+commit_sha: abcd1234efgh5678
+release_notes: Initial alpha release for CAMARA Fall26 release cycle.
+```
+
+## End-to-End Workflow
+
+### Step 1: Continuous Development on `main`
+
+- All PRs target the `main` branch, with changes validated via MegaLinter and YAML schema checks.
+- `release-plan.yaml` defines the roadmap and desired targets for the upcoming (pre-)release.
+- CI gates validate:
+  - Formatting and schema correctness
+  - Adherence to CAMARA guidelines based on the declared `release_status` and `api_status`
+  - Status progression logic (e.g., `rc` requires all APIs to be at least `rc`)
+- CI also raises non-blocking warnings about issues that must be addressed to enable promotion to the next stage (e.g., from `alpha` to `rc`, or `rc` to `release`).
+- All validation results are summarized in the CI output and posted as comments in the PR, helping developers stay informed and proactive.
+
+✅ Benefits:
+- Prevents invalid or incomplete content from entering `main`
+- Ensures developers align with release status expectations early
+- Encourages iterative, forward-looking quality improvements
+- Keeps readiness transparent across reviewers, contributors, and release managers
+- Allows optional enforcement that metadata updates (e.g., status changes) must be done in dedicated PRs separate from implementation changes
+
+### Step 2: Automated Release Branch Creation
+
+Upon triggering the release (e.g., via GitHub issue or label):
+
+- A release branch is created (e.g. `release/r4.1`)
+- A script or GitHub Action:
```

Comment:
We may go even further and instead of just replacing `wip` we may relay entirely on the CI to add certain common properties to the specs, e.g.

```
  license:
    name: Apache 2.0
    url: https://www.apache.org/licenses/LICENSE-2.0.html
  version: {{api_version}}
  x-camara-commonalities: {{commonalities_version}}

externalDocs:
  description: Product documentation at CAMARA
  url: https://github.com/camaraproject/{{working_group}}

servers:
  - url: "{apiRoot}/{{api_name}}/v{{url_api_version}}"
    variables:
      apiRoot:
        default: http://localhost:9091
        description: API root, defined by the service provider, e.g. `api.example.com` or `api.example.com/somepath`
```